docker/docker-compose.build.yml

```
@@ -0,0 +1,102 @@
+version: "3.4"
+
+services:
+
+  #
+  # TPL: Caffe 1.0
+  #
+  tpl-caffe-base:
+    # Base stage for common use by CPU and GPU build variants.
+    image: ${SMQTK_REGISTRY}/tpl-caffe:1.0-base
+    build:
+      context: tpl/caffe1
+      target: base
+      args:
+        CUDA_DEVEL_IMAGE_TAG:  # using value from .env file.
+
+  tpl-caffe-cpu:
+    # CPU-Only build variant
+    image: ${SMQTK_REGISTRY}/tpl-caffe:1.0-cpu
+    depends_on:
+      - tpl-caffe-base
+    build:
+      context: tpl/caffe1
+      target: caffe_cpu
+      args:
+        CUDA_DEVEL_IMAGE_TAG:  # using value from .env file.
+
+  tpl-caffe-gpu:
+    # GPU build variant
+    image: ${SMQTK_REGISTRY}/tpl-caffe:1.0-cuda${CUDA_DEVEL_IMAGE_TAG}
+    depends_on:
+      - tpl-caffe-base
+    build:
+      context: tpl/caffe1
+      target: caffe_gpu
+      args:
+        CUDA_DEVEL_IMAGE_TAG:  # using value from .env file.
+        # One or more CUDA architecture binary versions to compile for.
+        CUDA_ARCH_BIN_LIST: 52 60 61 62 70
```

Minor detail: A full suite of versions that I just tested as buildable with the 9.2 CUDA image base also includes `50` and `72`, so lets add that here:
```suggestion
        CUDA_ARCH_BIN_LIST: 50 52 60 61 62 70 72
```

This should be fixup committed into `4cd7d3e`.